### PR TITLE
Bug quarantine fix

### PR DESCRIPTION
**Pull request by blister**
I think the logical structure here is correct.

Need help with:

1. Remove role from all RSC servers.
2. Verifying any of this works in reality.
3. Figuring out what channel we're going to send "new messages" notifications to.

### CODE REVIEW

**Comment by BenjaminHunt** [comment]
A bit picky, but I wouldn't say "fix," I'd say something like "resolution" or "compromise" or "workaround"

File: dmHelper/dmHelper.py [hunk 1]
```diff
@@ -23,9 +26,26 @@ def __init__(self, bot):
         self.bot = bot
         self.message_queue : list = []
         self.priority_message_queue : list = []
+        self.errored_message_queue : list = [] # used to store DMs that were unable to be delivered
         self.actively_sending = False
+        self.bulk_role_manager_cog : BulkRoleManager = bot.get_cog("BulkRoleManager")
```
Comment (BenjaminHunt) [inline dmHelper/dmHelper.py:31]: Best to remove circular dependency. Adding roles is pretty simple, and I'd prefer to go with the role removal approach.

**Comment by BenjaminHunt** [comment]
So our solution should be:
```
- create role (you did, great!)
- create a command ?giveAllNeedsDMRole
- on listener for DMs:
  - remove needs DM role
  - send any saved messages
- on member join
  - if mutual guild --> we may want to patch this code on later
    - if other guild has needs dm role
      - if member has this role
        - add needs to dm bot role
```

**Review by BenjaminHunt — CHANGES_REQUESTED** [review]
A few things missing from our complete solution. I don't think we need 100% of it, but we should be adding the role on member join at minimum.

Might be good to localize the behavior to the guild just to get it working, then make a second PR for a full solution

File: dmHelper/dmHelper.py [hunk 2]
```diff
@@ -92,15 +112,26 @@ async def _process_dm_queues(self):
...
             
             if content or embed:
                 try:
-                    await recepient.send(content=content, embed=embed)
+                    await recipient.send(content=content, embed=embed)
                 except Exception as e:
                     message_data['exception'] = e
                     failed_msg_buffer.append(message_data)
-                    log.debug(f"DM to recepient \"{recepient.name}\" failed due to Exception: {e}")
+                    log.debug(f"DM to recipient \"{recipient.name}\" failed due to Exception: {e}")
```
[the remark this answers is not in the document]
Comment (blister) [reply dmHelper/dmHelper.py:122]: Yeah, I can do that. I didn't touch or turn off any of the existing code that was there.

File: dmHelper/dmHelper.py [hunk 3]
```diff
@@ -23,9 +27,25 @@ def __init__(self, bot):
         self.bot = bot
         self.message_queue : list = []
         self.priority_message_queue : list = []
+        self.errored_message_queue : list = [] # used to store DMs that were unable to be delivered
         self.actively_sending = False
         # self.task = asyncio.create_task(self.process_dm_queues())  # TODO: protect queue from bot crashes -- json?
     
+    @commands.Cog.listener('on_message')
+    async def _message_listener(self, message: Discord.Message)
```
Comment (BenjaminHunt) [inline dmHelper/dmHelper.py:35]:
```suggestion
    async def _message_listener(self, message: discord.Message):
```
We should also use message type to only listen to DMs
image

**Review by BenjaminHunt — CHANGES_REQUESTED** [review]

**Comment by blister** [comment]
Love your changes. Looks very clean now. Nice job finding the listener `on_message_without_command`.

**Review by BenjaminHunt — APPROVED** [review]
Good Teamwork

**Comment by BenjaminHunt** [comment]
## Message to other league admins

As you guys know, the bot was flagged as spam once again. This obviously blows, given the work we at the Dev Committee have put in to avoid this from happening on our new bot(s) but fortunately, the brilliant minds in the Dev Committee have devised what will likely be our best path forward.

### The Problem
The bot has now been "quarantined." We've discovered this _doesn't_ mean that it can't send DMs. It can't _initiate_ them. That's a big differentiation. So if players in the league DM the bot, then the bot can DM them back.

### Our Work-Around
The admins in the server must create a role, **Needs To DM Bot**, or some name of your choice. That's probably a good name so its consistent with RSC 3v3. This role will be assigned to all members within the server. You'll also need to create a new channel, (RSC 3v3 has **#please-dm-the-bot**. This channel's perms should be set so that nobody can send messages, and only Admins, Mods, Dev Committee, and Needs to DM Bot can view the channel.

This channel will have one concise message explaining the problem and that they need to DM the bot, so that it can DM them back. The bot will ghost ping players who need to DM the bot, so that they read that message and hopefully DM the bot, without clogging the channel for other players. This will be triggered whenever a player is assigned the role, either upon a failed DM, or a server join. When players DM the bot, it will remove this role. The admin commands to configure these channels after creation will be shared shortly.

We suggest this message:
> Hello! Unfortunately our RSC has been running into issues with the bots that are core to keeping this league running effectively. As a result, the bot can no longer send the first DM between any users.
> 
> We ask that you send the bot ANY message to enable DMs between you and the bot, and maximize your RSC experience!

### More Info
We will have some code in place to synchronize across all RSC servers, but we plan to improve it in a future iterations to cover edge cases better. Along with this, we intend to send previously failed messages when DMs are unlocked following this process.

If you have any questions, feel free to ping me. Just be aware that my responses may be a bit delayed as I am currently attending RLCS worlds.

We hope to get this solution out very soon.

**Pull request merged**